### PR TITLE
[Compute] `az disk create/update`: Fix `--encryption-type` processing

### PR DESCRIPTION
**Pull request by cxznmhdcxz**
**Related command**
See issue: https://github.com/Azure/azure-cli/issues/22439

Fix following performance:
1. `az disk create`:  if only specified `--encryption-type` but not `--disk-encryption-set`,  `--encryption-type` is actually ignored during processing
2. `az disk update`:  if only specified `--encryption-type EncryptionAtRestWithPlatformKey`, would not remove current disk encryption set(if the disk has it) and only update encryption type, thus may cause response error

**Description**

**Testing Guide**

**History Notes**

[Compute] `az disk create`: Fix the issue that specifying encryption type as `EncryptionAtRestWithPlatformKey` does not take effect when creating a disk
[Compute] `az disk update`: Fix the `(InvalidParameter) Resource xxx encrypted with platform key has disk encryption set id specified` error when updating the encryption type to platform managed keys 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Comment by cxznmhdcxz** [comment]
Right now this is the request body while trying to update a CMK disk's encryption type to PMK:
```
{"location": "eastus", "tags": {}, "sku": {"name": "Premium_LRS"}, "properties": {"hyperVGeneration": "V1", "creationData": {"createOption": "Empty"}, "diskSizeGB": 1, "diskIOPSReadWrite": 120, "diskMBpsReadWrite": 25, "encryption": {"type": "EncryptionAtRestWithPlatformKey"}, "networkAccessPolicy": "AllowAll", "tier": "P1", "publicNetworkAccess": "Enabled"}}
```


service will return this error:
```
(BadRequest) Cannot change encryption type of disk 'cd' from EncryptionAtRestWithCustomerKey to EncryptionAtRestWithPlatformKey
```

@grizzlytheodore  Is this by design?

**Review by wangzelin007 — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Could you add some tests for those change?

**Comment by VikasMohandoss** [comment]
Hi Xing,

With the feature registration, we should not be getting the error message.
Is this something we could test as well? There is no registration required
in Canary so it would be a good test env.

Thanks!
Vikas

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -373,6 +373,8 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
     encryption = None
     if disk_encryption_set:
         encryption = Encryption(type=encryption_type, disk_encryption_set_id=disk_encryption_set)
+    elif encryption_type:
+        encryption = Encryption(type=encryption_type)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:377]:
```suggestion
    if disk_encryption_set or encryption_type:
        encryption = Encryption(type=encryption_type, disk_encryption_set_id=disk_encryption_set)
```

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -464,8 +466,9 @@ def update_managed_disk(cmd, resource_group_name, instance, size_gb=None, sku=No
                 subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
                 namespace='Microsoft.Compute', type='diskEncryptionSets', name=disk_encryption_set)
         instance.encryption.disk_encryption_set_id = disk_encryption_set
-    if encryption_type is not None:
+    elif encryption_type is not None:
         instance.encryption.type = encryption_type
+        instance.encryption.disk_encryption_set_id = None
```
[the remark this answers is not in the document]
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:471]: @VikasMohandoss @laurawu19 @vikramd-ms Could you help confirm that when updating the disk, `disk_encryption_set` can be sent to service at the same time only when `encryption_type` is `EncryptionAtRestWithCustomerKey`. 
For other `encryption_type`, `disk_encryption_set` needs to be removed before requesting REST service. May I ask is my understanding right?

**Comment by zhoxing-ms** [comment]
@VikasMohandoss @laurawu19 @vikramd-ms Another question that needs your help to confirm: as mentioned by @cxznmhdcxz, when we update the `encryption_type` of disk from `EncryptionAtRestWithPlatformKey` to `EncryptionAtRestWithPlatformKey` to and clear the original `disk_encryption_set_id` , than service will return the following error message:

> (BadRequest) Cannot change encryption type of disk 'cd' from EncryptionAtRestWithCustomerKey to EncryptionAtRestWithPlatformKey

Does service currently not support updating the `encryption_type` of disk from `EncryptionAtRestWithCustomerKey` to `EncryptionAtRestWithPlatformKey`, right?
If so, does the this error message meet your expectations?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -464,6 +464,9 @@ def update_managed_disk(cmd, resource_group_name, instance, size_gb=None, sku=No
                 subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
                 namespace='Microsoft.Compute', type='diskEncryptionSets', name=disk_encryption_set)
         instance.encryption.disk_encryption_set_id = disk_encryption_set
+    if instance.encryption.type != 'EncryptionAtRestWithCustomerKey' and \
+                encryption_type != 'EncryptionAtRestWithCustomerKey':
+        instance.encryption.disk_encryption_set_id = None
     if encryption_type is not None:
         instance.encryption.type = encryption_type
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:471]:
```suggestion
    if encryption_type is not None:
        instance.encryption.type = encryption_type
        if instance.encryption.type != 'EncryptionAtRestWithCustomerKey' and \
                encryption_type != 'EncryptionAtRestWithCustomerKey':
            instance.encryption.disk_encryption_set_id = None
```

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 4]
```diff
@@ -466,6 +466,9 @@ def update_managed_disk(cmd, resource_group_name, instance, size_gb=None, sku=No
         instance.encryption.disk_encryption_set_id = disk_encryption_set
     if encryption_type is not None:
         instance.encryption.type = encryption_type
+        if instance.encryption.type != 'EncryptionAtRestWithCustomerKey' and \
+                encryption_type != 'EncryptionAtRestWithCustomerKey':
+            instance.encryption.disk_encryption_set_id = None
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:471]:
```suggestion
        if encryption_type != 'EncryptionAtRestWithCustomerKey':
            instance.encryption.disk_encryption_set_id = None
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**